test/gtest/kthvalue.hpp

```diff
@@ -0,0 +1,194 @@
+        KthvalueTestCase({3000, 8, 2, 4, 20}, 9, 0, false),  // 5D non-cont
+    };
+}
+
+template <typename TIO>
+struct KthvalueFwdTest : public ::testing::TestWithParam<KthvalueTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        config        = GetParam();
+
+        auto inDims    = config.GetDims();
+        auto inStrides = config.ComputeStrides(inDims);
+        if(config.dim < 0)
+        {
+            config.dim += inDims.size();
+        }
+        EXPECT_TRUE(config.dim >= 0 and config.dim < inDims.size());
+        auto outDims = config.GetDims();
+        if(!config.keepDim)
+        {
+            outDims.erase(outDims.begin() + config.dim);
+            if(outDims.empty())
+                outDims.push_back(1);
+        }
+        else
+        {
+            outDims[config.dim] = 1;
+        }
+
+        auto in_gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<TIO>(0.1, 200); };
+        input             = tensor<TIO>{inDims, inStrides}.generate(in_gen_value);
+
+        output = tensor<TIO>{outDims};
+        std::fill(output.begin(), output.end(), 0);
+
+        outputHost = tensor<TIO>{outDims};
+        std::fill(outputHost.begin(), outputHost.end(), 0);
+
+        // miopenDataType_t doesn't support size_t, I use double instead (both types use 64 bits)
+        indicesDesc       = miopen::TensorDescriptor(miopenDouble, outDims);
+        size_t outputSize = indicesDesc.GetElementSize();
+        indices.resize(outputSize);
+        indicesHost.resize(outputSize);
+
+        input_dev   = handle.Write(input.data);
+        indices_dev = handle.Write(indices);
+        output_dev  = handle.Write(output.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        miopenStatus_t status;
+
+        status = miopen::KthvalueForward(handle,
+                                         input.desc,
+                                         input_dev.get(),
+                                         output.desc,
+                                         output_dev.get(),
+                                         indicesDesc,
+                                         (size_t*)indices_dev.get(),
+                                         config.k,
+                                         config.dim,
+                                         config.keepDim);
+        cpu_kthvalue<TIO>(input, outputHost, indicesHost, indicesDesc, config.k, config.dim);
+
+        EXPECT_EQ(status, miopenStatusSuccess);
+        output.data = handle.Read<TIO>(output_dev, output.data.size());
+        indices     = handle.Read<size_t>(indices_dev, indices.size());
+    }
+
+    void Verify()
+    {
+        double threshold = std::numeric_limits<TIO>::epsilon();
+
+        auto error = miopen::rms_range(outputHost, output);
+
+        EXPECT_TRUE(miopen::range_distance(outputHost) == miopen::range_distance(output));
```

Comment:
you should also check indices result from indicesHost

Reply:
I did that I the first place. However, in case there are a lot of same values in the tensor. The return index by torch is unexpected. In this example, the input tensor contains only value 1. Since the kth value is computed by multithreads in the GPU, there is no guarantee what the return index of kth value may be. 
<img width="627" alt="image" src="https://github.com/moreh-dev/MIOpen/assets/52347285/b54adffb-34e3-4129-862c-737800d53b7d">